snapshots.go

```diff
@@ -0,0 +1,136 @@
+package godo
+
+import "fmt"
+
+const snapshotBasePath = "v2/snapshots"
+
+// SnapshotsService is an interface for interfacing with the snapshots
+// endpoints of the DigitalOcean API
+// See: https://developers.digitalocean.com/documentation/v2#snapshots
+type SnapshotsService interface {
+	List(*ListOptions) ([]Snapshot, *Response, error)
+	ListVolume(*ListOptions) ([]Snapshot, *Response, error)
+	ListDroplet(*ListOptions) ([]Snapshot, *Response, error)
+	Get(string) (*Snapshot, *Response, error)
+	Delete(string) (*Response, error)
+}
+
+// SnapshotsServiceOp handles communication with the snapshot related methods of the
+// DigitalOcean API.
+type SnapshotsServiceOp struct {
+	client *Client
+}
+
+var _ SnapshotsService = &SnapshotsServiceOp{}
+
+// Snapshot represents a DigitalOcean Snapshot
+type Snapshot struct {
+	ID            string   `json:"id,omitempty"`
```

Comment:
A little closer, I'm still seeing the same error. `ResourceType` is in the same situation, a string for volumes and an int for Droplets.

Here's the test program I'm using: https://gist.github.com/andrewsomething/d43ccaf333531001612647405188d4ac